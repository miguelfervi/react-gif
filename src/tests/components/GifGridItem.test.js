import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Test on <GifGridItem></GifGridItem>", () => {
  const title = "Un titulo";
  const url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should return the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should return the paragraph with the title", () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test("should return img equal to props", () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test("should return className", () => {
    const img = wrapper.find('div');

    const className = img.prop('className')
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
