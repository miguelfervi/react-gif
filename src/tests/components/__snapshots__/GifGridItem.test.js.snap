// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem></GifGridItem> should return the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
